File: flatform/src/main/java/nam/tran/flatform/IApi.java
package nam.tran.flatform;

import android.arch.lifecycle.LiveData;

import java.util.List;

import nam.tran.flatform.core.ApiResponse;
import nam.tran.flatform.model.response.session.SoccerSeason;
import nam.tran.flatform.model.response.team.ListTeamResponse;
import retrofit2.http.GET;
import retrofit2.http.Path;

public interface IApi {
    @GET("/v1/soccerseasons")
    LiveData<ApiResponse<List<SoccerSeason>>> getListSeason();

    @GET("v1/soccerseasons/{id}/teams")
    LiveData<ApiResponse<ListTeamResponse>> getListTeam(@Path("id") int id);
}

File: flatform/src/main/java/nam/tran/flatform/IApi.kt
package nam.tran.flatform

import android.arch.lifecycle.LiveData

import nam.tran.flatform.core.ApiResponse
import nam.tran.flatform.model.response.session.SoccerSeason
import nam.tran.flatform.model.response.team.ListTeamResponse
import retrofit2.http.GET
import retrofit2.http.Path

interface IApi {
    @get:GET("/v1/soccerseasons")
    val listSeason: LiveData<ApiResponse<List<SoccerSeason>>>

    @GET("v1/soccerseasons/{id}/teams")
    fun getListTeam(@Path("id") id: Int): LiveData<ApiResponse<ListTeamResponse>>
}
